awx_collection/plugins/modules/tower_job_launch.py

```diff
@@ -225,6 +224,9 @@ def main():
     for variable_name in check_vars_to_prompts:
         if module.params.get(variable_name) and not job_template[check_vars_to_prompts[variable_name]]:
             param_errors.append("The field {0} was specified but the job template does not allow for it to be overridden".format(variable_name))
+    # Check if Either ask_variables_on_launch, or survey_enabled is enabled for use of extra vars.
+    if module.params.get('extra_vars') and not (job_template['ask_variables_on_launch'] or job_template['survey_enabled']):
+        param_errors.append("The field extra_vars was specified but the job template does not allow for it to be overridden")
```

Comment:
My guidance on these cases is that we should rely on The API validation if possible. This doesn't do any harm, but it's obviously not comprehensive, and it would be burdensome to make it so. In other words, just don't validate inputs if the API is likely to give a descriptive message, and let the user see that message.

That's my general guiding principle for these cases, and mean it only as a suggestion. Again, this type of thing shouldn't do _harm_.

Reply:
Right now we could do that and I believe we would just need to change the assertions in our tests, however I don't believe it gives a good error message, and thought this was the best way to go around it our earlier restriction, while allowing everything the tower api allows through.